paramnb/__init__.py

```diff
@@ -197,7 +198,18 @@ def action_cb(button):
 
         def change_event(event):
             new_values = event['new']
-            setattr(self.parameterized, p_name, new_values)
+            if isinstance(w, ipywidgets.Text):
+                try:
+                    new_values = ast.literal_eval(new_values)
+                except:
+                    w.layout.border = '5px solid #cc0000'
+            try:
+                setattr(self.parameterized, p_name, new_values)
+            except ValueError:
+                w.layout.border = '5px solid #cc0000'
+                return
+            else:
+                w.layout.border = '0px'
```

Comment:
Given how arbitrary this looks at first, can the red-changing bits be wrapped in a function call with a good, clear name, and/or a parameter?